### ISSUE TITLE
Peer connector behavior during disconnects

### ISSUE THREAD

**Issue by richard-ramos** [waku-org/go-waku#425]
Currently the peer connector uses a [exponential backoff strategy](https://github.com/waku-org/go-waku/blob/master/waku/v2/node/wakunode2.go#L184-L188) to attempt to reconnect to peers. This works under normal conditions of network connection. 

However, in cases where the node loses connectivity, discV5 will keep sending peers since these will persist in routing tables for a while. These peers will then fail to connect, and end up with a bigger exponential backoff each time we attempt to reconnect to them, penalizing them for a problem that happens in our end. When the node goes back online it's possible that it wont have any peer to reconnect due to peers retry time existing too far in the future.

We need to implement some way to know if we have internet connection to inform the peer connector whether it has to try to connect to peers or not.

cc: @cammellos @iurimatias @rasom

**Comment by cammellos** [comment]
@richard-ramos there's a few ways to handle this.
Detecting connectivity is a bit fiddly, on mobile it works ok, on desktop is a bit harder (there's a `ConnectionChange` entry on geth backend that mobile uses when it knows that connection has changed, so that's already used in messenger and other parts (in wakuv2 `ConnectionChanged`), for exactly this reason https://github.com/status-im/status-go/blob/f85c500e9a112ad5a466e0d303aa0bc2d3776c8a/wakuv2/waku.go#L1405

Basically on connection change, we reset the backoff, but we check that the connection is not flapping, by setting a minimum.
Something like that should be safe enough I take?
